packages/core/useDevicePixelRatio/index.ts

```diff
@@ -34,21 +17,17 @@ export function useDevicePixelRatio({
     }
   }
 
-  const pixelRatio = ref(window.devicePixelRatio)
+  const pixelRatio = ref(1)
 
-  const handleDevicePixelRatio = () => {
+  let mqResolution: MediaQueryList
+  tryOnScopeDispose((function MqResolutionObserve(): Fn {
     pixelRatio.value = window.devicePixelRatio
-  }
-
-  useEventListener(window, 'resize', handleDevicePixelRatio, { passive: true })
-
-  DEVICE_PIXEL_RATIO_SCALES.forEach((dppx) => {
-    // listen mql events in both sides
-    const mqlMin = useMediaQuery(`screen and (min-resolution: ${dppx}dppx)`)
-    const mqlMax = useMediaQuery(`screen and (max-resolution: ${dppx}dppx)`)
-
-    watch([mqlMin, mqlMax], handleDevicePixelRatio)
-  })
+    mqResolution = window.matchMedia(`(resolution: ${pixelRatio.value}dppx)`)
+    mqResolution.addEventListener('change', MqResolutionObserve, { once: true })
+    return () => {
+      mqResolution.removeEventListener('change', MqResolutionObserve)
+    }
+  })())
```

Comment:
What's the reason for wraping the function like this?

Reply:
Just a short way to immediately invoke the function and passing `removeEventListener` to `tryOnScopeDispose`